flow/diff_context.cc

```diff
@@ -139,8 +139,13 @@ Damage DiffContext::ComputeDamage(const SkIRect& accumulated_buffer_damage,
   frame_damage.roundOut(&res.frame_damage);
 
   SkIRect frame_clip = SkIRect::MakeSize(frame_size_);
-  res.buffer_damage.intersect(frame_clip);
-  res.frame_damage.intersect(frame_clip);
+  [[maybe_unused]]
+  bool ret1 = res.buffer_damage.intersect(frame_clip);
+  [[maybe_unused]]
+  bool ret2 = res.frame_damage.intersect(frame_clip);
+  if (!ret1 || !ret2) {
+    FML_LOG(ERROR) << "intersect was not handled: " << ret1 << ", " << ret2;
```

Comment:
@knopp this is the interesting case here where I wanted to either turn these into `FML_DCHECK(ret)` calls or write a test case that verifies that we do the right thing here.

I was unable to figure out how to write a test case that specifically tested this case (see my ifdef'd out test case in `diff_context_unittests.cc`), but when I put in the `FML_DCHECK` calls they got triggered in several flow layer tests.

Please advise how I can write a better test case in the unittests and which gets the wrong answer if we don't handle the return values here and I'll change these to deal with the return value. Typically I use `if (!r1.intersect(r2)) { r1.setEmpty(); }` if that will work in this case...

Reply:
Obviously the other tests in the `*_layer_unittests.cc` files do run into this case, but they aren't specifically trying to test it, so they may either be hitting this as a side effect of the way they're written to test something else, or they are bad tests for what they are really trying to do. In either case, I don't want to rely on them to test this particular change...

Reply:
Looking further into the cases where the other unit tests were triggering this code, in all cases it was because the damage rects were already empty so ignoring the return value in that case didn't matter as the result rects were already empty and did not "need" to be set to empty again.

So, I'd like to be able to rule this out or find a test case that can trigger it without the damage rects being "already empty".

Reply:
BTW, I fixed the test code and found a way to trigger this problem so I just need a review now.